BankWallet/BankWallet/Modules/Restore/RestoreModule.swift

```diff
@@ -1,11 +1,10 @@
 protocol IRestoreView: class {
+    func set(predefinedAccountTypes: [IPredefinedAccountType])
 }
 
 protocol IRestoreViewDelegate {
     func viewDidLoad()
-    var typesCount: Int { get }
-    func type(index: Int) -> IPredefinedAccountType
-    func didSelect(index: Int)
+    func didSelect(accountType: IPredefinedAccountType)
```

Comment:
View layer should not pass any objects back to presenter. It can only provide clicked index.